roles/ceph-rgw-loadbalancer/tasks/pre_requisite.yml

```diff
@@ -1,11 +1,71 @@
 ---
+- name: assert that keepalived_version is defined
+  assert:
+    that:
+      - keepalived_version == 'distro' or keepalived_version is version('0.0.0', '>', strict=True)
+    fail_msg: keepalived_version must either be 'distro' or valid version number
+
 - name: install haproxy and keepalived
   package:
-    name: ['haproxy', 'keepalived']
+    name: haproxy
     state: present
   register: result
   until: result is succeeded
 
+- name: install keepalived
+  package:
+    name: keepalived
+    state: present
+  register: result
+  when: keepalived_version == 'distro'
+  until: result is succeeded
+
+- name: install non-distro version of keepalived
+  block:
+
+    - name: check if specified keepalived version already exists
+      command: "ls /usr/sbin/keepalived-{{ keepalived_version }}"
+      failed_when: false
+      changed_when: false
+      register: check_keepalived_version
+
+    - name: install required packages for keepalived
+      apt:
+        name: ['curl', 'gcc', 'libssl-dev', 'libnl-3-dev', 'libnl-genl-3-dev', 'libsnmp-dev']
+        state: present
+
+    - name: unarchive keepalived
+      unarchive:
+        src: "{{ keepalived_url }}"
+        dest: /tmp/
+        remote_src: true
+        owner: root
+        group: root
+
+    - name: build keepalived
+      command: "./configure --prefix=/usr/sbin/keepalived-{{ keepalived_version }}"
+      args:
+        chdir: "/tmp/keepalived-{{ keepalived_version }}/"
+      when:
+        - check_keepalived_version.rc != 0
+
+    - name: install keepalived
+      make:
+        chdir: "/tmp/keepalived-{{ keepalived_version }}/"
+        target: install
+      when:
+        - check_keepalived_version.rc != 0
+
+    - name: update binary to updated keepalived version
+      file:
+        path: /usr/sbin/keepalived
+        src: "/usr/sbin/keepalived-{{ keepalived_version }}/sbin/keepalived"
+        state: link
+        force: true
+  when:
+    - keepalived_version is defined
```

Comment:
The assertions being set ensure that `keepalived_version` is defined, so can drop this when condition.

Reply:
removed